SCAutolib/models/gui.py

```diff
@@ -0,0 +1,355 @@
+        return filename
+
+
+class Mouse():
+    """Controls the mouse of the system under test
+    """
+
+    def __init__(self):
+        run(['modprobe', 'uinput'], check=True)
+
+        # Maximum coordinate for both axis
+        self.ABS_MAX = 2**16
+
+        # initialize the uinput device
+        self.device = uinput.Device((
+            uinput.ABS_X + (0, self.ABS_MAX, 0, 0),
+            uinput.ABS_Y + (0, self.ABS_MAX, 0, 0),
+            uinput.BTN_LEFT,
+            uinput.BTN_MIDDLE,
+            uinput.BTN_RIGHT,
+            uinput.REL_WHEEL,
+        ))
+
+        self.CLICK_HOLD_TIME = 0.1
+
+    def move(self, x: float, y: float):
+        """Moves the mouse cursor to specified absolute coordinate.
+        Both coordinates are float numbers in range from 0 to 1.
+        These get mapped to the screen resolution in the compositor.
+
+        :param x: X coordinate of the cursor
+        :param y: Y coordinate of the cursor
+        """
+
+        logger.info(f'Moving mouse to {x, y})')
+
+        for uinput_axis, value in [(uinput.ABS_X, x), (uinput.ABS_Y, y)]:
+            # Check if value between 0 and 1
+            assert ((value >= 0) and (value <= 1))
+            converted = int(value * self.ABS_MAX)
+            self.device.emit(uinput_axis, converted, syn=False)
+
+        # Both axis move at once
+        self.device.syn()
+
+    def click(self, button: str = 'left'):
+        """Clicks any button of the mouse.
+
+        :param button: mouse button to click, defaults to 'left'
+            Possible values 'left', 'right' or 'middle'.
+        """
+
+        button_map = {
+            'left': uinput.BTN_LEFT,
+            'right': uinput.BTN_RIGHT,
+            'middle': uinput.BTN_MIDDLE,
+        }
+        uinput_button = button_map[button]
+
+        logger.info(f'Clicking the {button} mouse button')
+
+        # press the button
+        self.device.emit(uinput_button, 1)
+        # wait a little
+        sleep(self.CLICK_HOLD_TIME)
+        # release the button
+        self.device.emit(uinput_button, 0)
+
+
+def image_to_data(filename):
+    """
+    Convert screenshot into dataframe of words with their coordinates.
+    """
+    UPSCALING_FACTOR = 2
+
+    image = cv2.imread(filename)
+    grayscale = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
+    upscaled = cv2.resize(grayscale,
+                          dsize=None,
+                          fx=UPSCALING_FACTOR,
+                          fy=UPSCALING_FACTOR,
+                          interpolation=cv2.INTER_LANCZOS4)
+    _, binary = cv2.threshold(upscaled, 120, 255, cv2.THRESH_BINARY_INV)
+    df = pytesseract.image_to_data(binary, output_type='data.frame')
+
+    yres, xres = binary.shape[:2]
+    df[['left', 'width']] /= xres
+    df[['top', 'height']] /= yres
+
+    logger.debug(df)
+    return df
+
+
+def images_same(filename1: str, filename2: str):
+    """Compare two images, return True if they are completely identical."""
```

Comment:
Parameters description is missing. Image is too general, does it eat .jpg or .png or both or none? I don't like name of this function,  images_equals or images_are_same would be better.